### ISSUE TITLE
Client registration error in IAM HA when grant_type is 'device'

### ISSUE THREAD

**Issue by federicaagostini** [indigo-iam/iam#440]
We have an issue when we register a new client with oidc-agent (using the device OIDC flow) in an IAM HA instance.
In particular when we copy-and-paste the device code into IAM endpoint we got stuck.

In attachment the iam-login-service log: [iam-login-service_iam-ha.log](https://github.com/indigo-iam/iam/files/7529440/iam-login-service_iam-ha.log)

**Comment by federicaagostini** [comment]
Deployed an [IAM dev](https://iam-dev.cloud.cnaf.infn.it) instance where the session store type is set to be Redis.  
Not yet HA, just one replica.

The device code flow looks like working here

```
$ oidc-gen -w device dev-device
[1] https://wlcg.cloud.cnaf.infn.it/
[2] https://iam-dev.cloud.cnaf.infn.it/
[3] https://iam.cnaf.infn.it/
[4] https://iam-escape.cloud.cnaf.infn.it/
[5] https://lhcb-auth.web.cern.ch/
[6] http://localhost:8080/
[7] https://iam-test.indigo-datacloud.eu/
[8] https://iam.deep-hybrid-datacloud.eu/
[9] https://iam.extreme-datacloud.eu/
[10] https://iam-demo.cloud.cnaf.infn.it/
[11] https://b2access.eudat.eu:8443/oauth2
[12] https://b2access-integration.fz-juelich.de/oauth2
[13] https://login-dev.helmholtz.de/oauth2
[14] https://login.helmholtz.de/oauth2
[15] https://services.humanbrainproject.eu/oidc/
[16] https://accounts.google.com
[17] https://aai.egi.eu/oidc/
[18] https://aai-dev.egi.eu/oidc/
[19] https://aai-dev.egi.eu/auth/realms/egi
[20] https://aai-demo.egi.eu/auth/realms/egi
[21] https://aai.egi.eu/auth/realms/egi
[22] https://login.elixir-czech.org/oidc/
[23] https://oidc.scc.kit.edu/auth/realms/kit
Issuer [https://wlcg.cloud.cnaf.infn.it/]: https://iam-dev.cloud.cnaf.infn.it/
The following scopes are supported: openid profile email address phone offline_access eduperson_scoped_affiliation eduperson_entitlement wlcg wlcg.groups test-scope aarc eduperson_assurance
Scopes or 'max' (space separated) [openid profile offline_access]: max
Registering Client ...
Generating account configuration ...
accepted

Using a browser on any device, visit:
https://iam-dev.cloud.cnaf.infn.it/device

And enter the code: AGZDTJ
Alternatively you can use the following QR code to visit the above listed URL.

█████████████████████████████████████
██ ▄▄▄▄▄ █ ██▀▀  ▀▄▀▀▀█▄█▄▀█ ▄▄▄▄▄ ██
██ █   █ █  ▀█ █▄█▀▄ █ ▄██▄█ █   █ ██
██ █▄▄▄█ █▀  █▄█████▄▀█ ▀▄ █ █▄▄▄█ ██
██▄▄▄▄▄▄▄█▄█ ▀▄▀▄▀ ▀ ▀ ▀ █ █▄▄▄▄▄▄▄██
██▄ █▀ ▄▄▀▀█▄█▄ ▀▀▄▄██▀▄▄▀▄▀█▄█▄ ▄▄██
███▄▄█ █▄█▄ ▄█▀█▀▀▄▀██ ▀▀ ▄▄ ▄▀▀▄▄███
██   ▀▀█▄▀ █▀▄▀▄▄█▄▀█▄▄ ▄▀▄▀▄ ▄▄█▄▄██
██▀ ▀▀▀█▄█▀▄  ▄ ▄█▄▀ ▀▄▀▀▄▀▀▀█  ▄▄███
██▄▄█▀█ ▄▀█▀██▄ ▀█▀▄▀▄▄ █ ▄▀▄▀▄▀██▄██
██ ▀▀▄█▀▄▄█▀▄█▀█▀█ ▀▀█▄▀▀▄▄▄▄▄ ▀▄▄███
██ ▄▀█ ▄▄ ▄ ▄▄▀▄▄█  ▀█▄ ▄▀▄▀▄▀▄▀██▄██
████ █▄█▄█▄█▄ ▄ ▄▀▀▀ ▀▀ ▄▄▀  █▄█▄▄███
██▄▄▄▄▄▄▄▄▀▀ █▄ ▀▀▄▄▀▄█▄ ▄ ▄▄▄ ██ ███
██ ▄▄▄▄▄ █▀▄ █▀█▀   ▄█  ▀▀ █▄█ ██▄███
██ █   █ █▄█▀▄▀▄▄▀█▄▀▄▄ ██ ▄ ▄ █▀█▄██
██ █▄▄▄█ █▀   ▄ ▄ █  █▀▀█▀▀ ▄▄ ██████
██▄▄▄▄▄▄▄█▄█▄█▄▄█▄▄▄▄▄▄▄▄▄▄█▄▄▄███▄██
█████████████████████████████████████

Enter encryption password for account configuration 'dev-device': 
Confirm encryption Password: 
Everything setup correctly!

$ oidc-token dev-device | cut -d. -f2 | base64 -d 2>/dev/null | jq
{
  "wlcg.ver": "1.0",
  "sub": "fec47318-f6b0-4cb1-9969-17f839b685c3",
  "aud": "https://wlcg.cern.ch/jwt/v1/any",
  "nbf": 1667492324,
  "scope": "address openid eduperson_entitlement wlcg phone offline_access test-scope eduperson_scoped_affiliation eduperson_assurance aarc wlcg.groups",
  "iss": "https://iam-dev.cloud.cnaf.infn.it/",
  "exp": 1667495924,
  "iat": 1667492324,
  "jti": "30823581-7b7e-4e03-9d13-3d8387cab123",
  "client_id": "cf1b0d18-0c3d-440d-9d42-c847ad8546c1",
  "wlcg.groups": [
    "/dev",
    "/test.vo",
    "/test.vo/G1"
  ]
}
```

**Comment by rmiccoli** [comment]
Solved by #535 in IAM v1.8.0
